eng/pipelines/extra-platforms/runtime-extra-platforms-ioslikesimulator.yml

```diff
@@ -45,3 +45,45 @@ jobs:
         creator: dotnet-bot
         interpreter: true
         testRunNamePrefixSuffix: Mono_$(_BuildConfig)
+
+#
+# Build the whole product using Mono for iOSSimulator/tvOSSimulator and run runtime tests with iOS/tvOS simulators
+#
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/common/global-build-job.yml
+    helixQueuesTemplate: /eng/pipelines/coreclr/templates/helix-queues-setup.yml
+    buildConfig: Release
+    runtimeFlavor: mono
+    isExtraPlatformsBuild: ${{ parameters.isExtraPlatformsBuild }}
+    isiOSLikeOnlyBuild: ${{ parameters.isiOSLikeOnlyBuild }}
+    platforms:
+      - iossimulator_x64
+      - tvossimulator_x64
+      - ${{ if eq(variables['isRollingBuild'], true) }}:
+        - iossimulator_arm64
```

Comment:
Good question, not sure. I will check it. 

Does other ioslike pipelines test the `iossimulator_arm64`? I have been mostly working with `iossimulator_x64` and `ios-arm64`.

Reply:
Its on the rolling build, but this PR won't trigger the rolling build.
I think its worth removing the condition for one build to see if its working as expected, and then adding the condition back in if its working.

Reply:
The tests are failing on `iossimulator-arm64`. @mdh1418 Could the following be a test-related failure?
```
tracing/eventpipe/bigevent/bigevent/bigevent.sh [FAIL]
  System.AggregateException : One or more errors occurred. (Failed to install mobile app.
  Expected: True
  Actual:   False) (The following constructor parameters did not have matching fixture data: _Global globalVar)
  ---- Failed to install mobile app.
  Expected: True
  Actual:   False
  ---- The following constructor parameters did not have matching fixture data: _Global globalVar
  Stack Trace:
    
    ----- Inner Stack Trace #1 (Xunit.Sdk.TrueException) -----
       at tracing_eventpipe._Global..ctor()
       at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
       at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
    ----- Inner Stack Trace #2 (Xunit.Sdk.TestClassException) -----
```

Reply:
Looking at the install related logs at https://helix.dot.net/api/jobs/6ccdfdb4-2b79-4887-8d88-cbc88faa7f8d/workitems/PayloadGroup0?api-version=2019-06-17 (got to here from L25 `Sent Helix Job; See work items at ` from the build)
It seems like the culprit was that there were no ios simulators arm64 available on the `ios-simulator-64` target?

Reply:
If there would be no simulator (or Xcode), it should say so and finish fast (return `NO_DEVICE_FOUND` or something). Timing out when querying simulators might be a problem between mlaunch and Xcode somewhere (maybe because of arm?).

In the logs from listing devices I see this:
```
[02:14:55.9250240] Simulator listing finished (47 kB)
[02:14:55.9757320] 
[02:14:55.9757520] Running /Applications/Xcode131.app/Contents/Developer/usr/bin/simctl create "iPhone X (iOS 15.0) - created by XHarness" com.apple.CoreSimulator.SimDeviceType.iPhone-X com.apple.CoreSimulator.SimRuntime.iOS-15-0
[02:14:56.2428170] Install Started
[02:15:56.0216660] Process 89120 didn't exit within 00:01:00 and will be killed
[02:15:56.0312070] Killing process tree of 89120...
[02:15:56.0924180] Pids to kill: 89120, 89123
[02:15:56.0928950] Running lldb diagnostics for pid 89120
[02:15:56.0947810] Printing backtrace for pid=89120
[02:15:58.3818000] (lldb) command source -s 0 '/tmp/helix/working/BE020A59/t/tmpl4xcFp.tmp'
[02:15:58.3820990] Executing commands in '/tmp/helix/working/BE020A59/t/tmpl4xcFp.tmp'.
[02:15:58.3821160] (lldb) process attach --pid 89120
[02:16:16.3255990] Running lldb diagnostics for pid 89123
[02:16:16.3259770] Printing backtrace for pid=89123
[02:16:16.3830710] (lldb) command source -s 0 '/tmp/helix/working/BE020A59/t/tmpbtSUMm.tmp'
[02:16:16.3839250] Executing commands in '/tmp/helix/working/BE020A59/t/tmpbtSUMm.tmp'.
[02:16:16.3839390] (lldb) process attach --pid 89123
[02:16:36.5565030] 89120 Execution timed out after 60 seconds and the process was killed.
[02:16:36.5653810] Process simctl exited with 137
[02:16:36.5693990] Could not create device
runtime: com.apple.CoreSimulator.SimRuntime.iOS-15-0
device type: com.apple.CoreSimulator.SimDeviceType.iPhone-X
[02:16:36.6211240] Failed to find/create simulator (attempt 1/3):
Microsoft.DotNet.XHarness.Common.CLI.NoDeviceFoundException: Could not create device
runtime: com.apple.CoreSimulator.SimRuntime.iOS-15-0
device type: com.apple.CoreSimulator.SimDeviceType.iPhone-X
   at Microsoft.DotNet.XHarness.iOS.Shared.Hardware.SimulatorLoader.FindOrCreateDevicesAsync(ILog log, String runtime, String devicetype, Boolean force, CancellationToken cancellationToken) in /_/src/Microsoft.DotNet.XHarness.iOS.Shared/Hardware/SimulatorLoader.cs:line 232
   at Microsoft.DotNet.XHarness.iOS.Shared.Hardware.SimulatorLoader.FindSimulators(TestTargetOs target, ILog log, Boolean createIfNeeded, Boolean minVersion, CancellationToken cancellationToken) in /_/src/Microsoft.DotNet.XHarness.iOS.Shared/Hardware/SimulatorLoader.cs:line 412
   at Microsoft.DotNet.XHarness.iOS.Shared.Hardware.SimulatorLoader.FindSimulators(TestTargetOs target, ILog log, Int32 retryCount, Boolean createIfNeeded, Boolean minVersion, CancellationToken cancellationToken) in /_/src/Microsoft.DotNet.XHarness.iOS.Shared/Hardware/SimulatorLoader.cs:line 518
[02:16:36.6226130] 
[02:16:36.6226260] Running /Applications/Xcode131.app/Contents/Developer/usr/bin/simctl create "iPhone X (iOS 15.0) - created by XHarness" com.apple.CoreSimulator.SimDeviceType.iPhone-X com.apple.CoreSimulator.SimRuntime.iOS-15-0
[02:16:36.8709390] Install Started
```

https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-79169-merge-6ccdfdb42b7948878d/PayloadGroup0/1/Reports/install/list-ios-simulator-64-20230417_021451.log?helixlogtype=result

Reply:
Ithink this line is the problem
```
[02:16:36.5693990] Could not create device
```

Reply:
Does this happen on a re-run?

Reply:
Rerunning now, noting the machine that failed to create the device: `machine dci-macm1-build-068.local running macOS-12.4`